File: Social medie projekt/Client/src/app/_models/post.ts
import { User } from "./user"

export interface Post {
    postId: number;
    userId: number;
    title: string;
    date: string;
    desc?: string;
    pictureURL?: string;
    tag?: Array<string>;

    // ??
    likedByUser: boolean;

    user: User;
}

File: Social_medie_projekt/WebApi/Database/Entities/Post.cs
namespace WebApi.Database.Entities
{
    public class Post
    {
        [Key]
        public int PostId { get; set; }

        [ForeignKey("User.UserId")]
        public int UserId { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string Title { get; set; } = string.Empty;

        [Column(TypeName = "nvarchar(1000)")]
        public string Desc { get; set; } = string.Empty;

        [Column(TypeName = "datetime")]
        public DateTime Date { get; set; } = DateTime.Now;

        [Column(TypeName = "bit")]
        public bool IsDeleted { get; set; }

        public PostLikes? PostLikes { get; set; }

        public User? User { get; set; }
    }
}
